src/v/cluster/topics_frontend.cc

```diff
@@ -97,6 +100,25 @@ needs_linearizable_barrier(const std::vector<topic_result>& results) {
 ss::future<std::vector<topic_result>> topics_frontend::create_topics(
   std::vector<custom_assignable_topic_configuration> topics,
   model::timeout_clock::time_point timeout) {
+    for (auto& tp : topics) {
+        if (!tp.cfg.properties.shadow_indexing.has_value()) {
+            tp.cfg.properties.shadow_indexing
+              = _metadata_cache.get_default_shadow_indexing_mode();
```

Comment:
nit: since a manual topic already goes through this assignment in https://github.com/redpanda-data/redpanda/blob/6e70513d845094c5419765c19968ebfbbeed555b/src/v/kafka/server/handlers/topics/types.cc#L168
I would add a comment here to acknowledge that